frontend/src/component/admin/users/UsersHeader/UsersHeader.tsx

```diff
@@ -23,8 +23,12 @@ const StyledElement = styled(Box)(({ theme }) => ({
 }));
 
 export const UsersHeader = () => {
-    const { isOss } = useUiConfig();
-    const licensedUsersEnabled = !isOss();
+    const {
+        isEnterprise,
+        uiConfig: { billing },
+    } = useUiConfig();
+
+    const licensedUsersEnabled = isEnterprise() && billing !== 'pay-as-you-go';
```

Comment:
shouldn't this be modelled in the backend code so that frontend doesn't have to do the multi conditional checking?